### PR TITLE
[DO NOT MERGE]: accessibility docs for SWC05

### PR DESCRIPTION
**Pull request by nikkimk**
----

## !!! NOTE TO REVIEWERS !!!

_Please **only** review the docs listed in **How has this been tested?** section._ 

When this PR is reviewed, it will be merged into #4756, where all of the changes will be reviewed before merging into main. 

_Please place **any other feedback** in #4756_ 

----

## Description

Improving the accessibility documentation of components.

## Related issue(s)

- #4756 

## Motivation and context

Documentation should provide more information and examples that demonstrate how to use the components accessibly.

### `<sp-icon>`

Review [icon docs](https://nikkimk-a11y-docs-iron-man-2--spectrum-web-components.netlify.app/components/icon/)

-  [ ] Are the examples accessible?
-  [ ] Do the docs examples and text provide information on how to use the component accessibly?
-  [ ] If the component is to be used in the context of another component, do the examples include how that component is used accessibly in that context?
- [ ] Are the docs headings logical and consistent across these components? ([1](#wave))
    

### `<sp-icons>`

Review [icons docs](https://nikkimk-a11y-docs-iron-man-2--spectrum-web-components.netlify.app/components/icons/)

-  [ ] Are the examples accessible?
-  [ ] Do the docs examples and text provide information on how to use the component accessibly?
-  [ ] If the component is to be used in the context of another component, do the examples include how that component is used accessibly in that context?
- [ ] Are the docs headings logical and consistent across these components? ([1](#wave))
    
### `<sp-iconset>`

Review [iconset docs](https://nikkimk-a11y-docs-iron-man-2--spectrum-web-components.netlify.app/components/iconset/)

-  [ ] Are the examples accessible?
-  [ ] Do the docs examples and text provide information on how to use the component accessibly?
-  [ ] If the component is to be used in the context of another component, do the examples include how that component is used accessibly in that context?
- [ ] Are the docs headings logical and consistent across these components? ([1](#wave))

### `<sp-icons-ui>` 

Review [icons-ui docs](https://nikkimk-a11y-docs-iron-man-2--spectrum-web-components.netlify.app/components/icons-ui/)

-  [ ] Are the examples accessible?
-  [ ] Do the docs examples and text provide information on how to use the component accessibly?
-  [ ] If the component is to be used in the context of another component, do the examples include how that component is used accessibly in that context?
- [ ] Are the docs headings logical and consistent across these components? ([1](#wave))

### `<sp-icons-workflow>`

Review [icons-workflow docs](https://nikkimk-a11y-docs-iron-man-2--spectrum-web-components.netlify.app/components/icons-workflow)

-  [ ] Are the examples accessible?
-  [ ] Do the docs examples and text provide information on how to use the component accessibly?
-  [ ] If the component is to be used in the context of another component, do the examples include how that component is used accessibly in that context?
- [ ] Are the docs headings logical and consistent across these components? ([1](#wave))

### Developing a Component Guide

- [ ] Does the [guide for developing a component](https://nikkimk-a11y-docs--spectrum-web-components.netlify.app/guides/adding-component/#documenting-the-component) provide enough information to create accessible, consistent component docs? ([1](#wave))

<a id="wave">1.</a> You can use the [WAVE browser extension's](https://wave.webaim.org/extension/) **Structure** tab to review heading structure.

### Font sizes

- [ ] Are the above component document font sizes consistent with the following design recommendations?
  - `h1`: heading XXL, serif, heavy
  - `h2`: heading L, sans-serif
  - `h3`: heading M, sans-serif
  - `h4`: heading S, sans-serif
  - `h5`: detail L
  - `body`: body M, sans-serif

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes. **_N/A_**
-   [ ] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Review by Rajdeepc — CHANGES_REQUESTED** [review]
Thanks for doing this! This will definitely help improve the documentation guidelines for our library. Though I see issues with the css on the tabs. It would be great if you could take a look at those before we do an official review!

**Comment by nikkimk** [comment]
> Though I see issues with the css on the tabs. It would be great if you could take a look at those before we do an official review!

@Rajdeepc Can you be more specific about the issues with tabs please? Do you mean the Overview /API / Changelog tabs, the Tabs component documentation, or something else?  Thanks!

**Comment by Rajdeepc** [comment]
> * Documentation Site
Please review the changes. The Docs site is distorted. Check H1 and H2!

Screenshot 2024-10-08 at 10 16 39 AM

**Comment by nikkimk** [comment]
@Rajdeepc visual changes and heading hierarchy changes were intentional for accessibility. The heading markup reflects the content hierarchy now. Heading font sizes for all headings have increase so that subheadings are also accessible.

File: packages/icons/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## Description
+## Overview
 
 The `<sp-icons-medium>` and `<sp-icons-large>` elements that are included in this package supply your application with the Spectrum CSS Icons at both the medium and large sizes for use via the `<sp-icon>` element also provided by the Spectrum Web Components library. Include at least one of these elements in a project that makes use of icons in these sets, but feel free to include these sets in the scope of any element that leverages them as they will be deduplicated as appropriate to ensure all of your components are able to deliver the icons included therein.
```
Comment (najikahalsema) [inline packages/icons/README.md:3]:
```suggestion
The `<sp-icons-medium>` and `<sp-icons-large>` elements included in this package supply your application with the Spectrum CSS medium and large icons for use in the `<sp-icon>` element. Include at least one of these elements in a project that makes use of icons in these sets. You can also include these sets in the scope of any element that leverages them, as they will be deduplicated to ensure all of your components can deliver the icons included therein.
```

**Review by najikahalsema — CHANGES_REQUESTED** [review]
Looking good! Just a few things.

File: packages/action-menu/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-## Description
+## Overview
 
 An `<sp-action-menu>` is an action button that triggers an overlay with `<sp-menu-items>` for activation. Use an `<sp-menu>` element to outline the items that will be made available to the user when interacting with the `<sp-action-menu>` element. By default `<sp-action-menu>` does not manage a selection. If you'd like for a selection to be held by the `<sp-menu>` that is presented in its overlay, use `selects="single"` to activate this functionality.
```
[the remark this answers is not in the document]
Comment (nikkimk) [reply packages/action-menu/README.md]: Trigger happy is fine by me :)

File: packages/help-text/README.md [hunk 3]
```diff
@@ -23,67 +23,138 @@ When looking to leverage the `HelpText` base class as a type and/or for extensio
...
+        Create a password with at least 8 characters.
+    </sp-help-text>
+</sp-textfield>
 ```
 
 </sp-tab-panel>
 </sp-tabs>
 
-## Negative
+#### Negative
```
[the remark this answers is not in the document]
Comment (nikkimk) [reply packages/help-text/README.md:85]: I renamed Variants to Options to better align with design's documentation

**Comment by nikkimk** [comment]
@Rajdeepc got the following design recommendations and implemented them:

- h1: heading XXL, serif, heavy
- h2: heading L, sans-serif
- h3: heading M, sans-serif
- h4: heading S, sans-serif
- h5: detail L

Please take another look, keeping in mind that the scope of review should be limited to the heading changes and the documentation listed above. 

**Note:** _This PR (originally for sprint SWC05) will get merged into the #4756 where all the docs will be getting a final review._

**Review by najikahalsema — APPROVED** [review]
LGTM!! Thank you!!!

**Review by Rajdeepc — APPROVED** [review]

**Pull request closed without merging**